src/item.cpp

```diff
@@ -5015,3 +5031,89 @@ bool item_category::operator!=( const item_category &rhs ) const
 {
     return !( *this == rhs );
 }
+
+/*-----------------------------------------------------------------------------
+ *  storage functions
+ *-----------------------------------------------------------------------------*/
+typedef std::vector<item>::iterator                                item_iterator;
+typedef std::vector<item>::const_iterator                    item_const_iterator;
+
+storage::storage(item *self, item_iterator start, item_iterator stop)
+{
+    me = self;
```

Comment:
Linking back to the parent object is *usually* not a good approach. This implementation will give you crashes as the pointer is not updated by the copy constructor nor by the assignment operator.
```C++
item a = ... ; // a.content.me == &a
item b = ... ; // b.content.me == &b
a = b; // a.content.me == &b
```